service_discovery/proto/core/v1/service_registry.proto

```diff
@@ -0,0 +1,107 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+// SPDX-License-Identifier: MIT
+
+syntax = "proto3";
+package service_registry;
+
+// Service Registry definition
+//
+// The protobuf definitions for Chariott's service registry
+
+// The entry point for the Registry gRPC Service.
+service ServiceRegistry {
+  // Register, or add a service to the registry
+  rpc Register(RegisterRequest) returns (RegisterResponse) {}
+
+  // Unregister, or remove a service from the registry
+  rpc Unregister(UnregisterRequest) returns (UnregisterResponse) {}
+
+  // Discover, or retrieve the metadata for a single service given its fully qualified name
+  rpc Discover(DiscoverRequest) returns (DiscoverResponse) {}
+
+  // Discover a list of services given their namespace
+  rpc DiscoverByNamespace(DiscoverByNamespaceRequest) returns (DiscoverByNamespaceResponse) {}
+
+  // List, or retrieve all contents of the service registry
+  rpc List(ListRequest) returns (ListResponse) {}
+}
+
+// Representation of a service including all of its metadata that the registry stores
+message ServiceMetadata {
+  // Logical grouping of services. Multiple services can share the same namespace.
+  // example: sdv.samples
+  string namespace = 1;
+  // Unique name of the service.
+  string name = 2;
+  // The combination of namespace, name, and version uniquely identify a service
+  string version = 3;
+  // The uri that clients can use to communicate with this service
+  string uri = 4;
+  // Used by clients to validate that they can communicate with this service. A short description of
+  // the communication kind, potentially including the network protocol and api specification type.
+  // example: grpc+proto
+  string communication_kind = 5;
+  // Used by clients to validate that they can communicate with this service. Communication communication_reference
+  // can be a reference to the api specification
+  // example: hello_world_service.v1.proto
+  string communication_reference = 6;
+}
+
+// Status of a registration operation
+enum RegistrationStatus {
+  // An entry did not previously exist in the service registry for this service
+  NEWLY_REGISTERED = 0;
```

Comment:
do we need this? now that we failed when something is already register

Reply:
I left it in originally because I anticipate that we will need to add it back at some point, but to keep it more consistent with the other APIs I will remove it.